### PR TITLE
Add fishbowl to update version workflow

### LINKED ISSUE #1014: Automate core gafferpy maintenance with fishbowl
Fishbowl was integrated as a sub-module to gafferpy in #981. It is now used to generate the core api of gafferpy, but this must be done manually after every Gaffer release and before every gaffer-tools release.

To use fishbowl to generate every supported Gaffer Operation, the spring-rest api must be used. This is because it contains every store on its [classpath](https://github.com/gchq/Gaffer/blob/df580eb9636b5649e6c182c617a555efd33aa83e/rest-api/spring-rest/pom.xml#L64) and has an extra [endpoint](https://github.com/gchq/Gaffer/blob/df580eb9636b5649e6c182c617a555efd33aa83e/rest-api/spring-rest/src/main/java/uk/gov/gchq/gaffer/rest/controller/IOperationController.java#L65) to get details of those Operations, even if they aren't supported by the current running store.
However, there is no simple way to get Gaffer's spring-rest running within the gaffer-tools repository.

The [gaffer-rest](https://hub.docker.com/r/gchq/gaffer-rest) docker image could be used, but right now this is built in the gaffer-docker release stage. This happens after a gaffer-tools release, as it relies on the new release of the ui, so there is a dependency order issue.
If we decided to build an image as part of a Gaffer release, then perhaps this could be used. This has been proposed in https://github.com/gchq/Gaffer/issues/2694.

### PR DESCRIPTION
**Pull request by t92549**
# Related Issue

- Resolve #1014

### CODE REVIEW

File: .github/workflows/update-gaffer-version.yaml [hunk 1]
```diff
@@ -14,32 +14,40 @@ jobs:
...
+        curl -o spring-rest.jar https://repo.maven.apache.org/maven2/uk/gov/gchq/gaffer/spring-rest/$newVersion/spring-rest-$newVersion-exec.jar
+        java -Dgaffer.schemas=ui/example/road-traffic/schema/ -Dgaffer.storeProperties=ui/example/federated/federatedStore.properties -Dgaffer.graph.config=ui/example/federated/graphConfig.json -jar spring-rest.jar &
+        sleep 1m
+        python python-shell/src/generate.py
+        rm spring-rest.jar
 
     - name: Submit PR
-      if: ${{ success() }}
-      uses: repo-sync/pull-request@v2
+      uses: peter-evans/create-pull-request@v4
```
Comment (GCHQDeveloper314) [inline .github/workflows/update-gaffer-version.yaml:49]: I assume this means the user creating the PR will now be GitHub Actions and not the "Gaffer" user?

File: .github/workflows/update-gaffer-version.yaml [hunk 2]
```diff
@@ -14,32 +14,40 @@ jobs:
...
+        sed -i'' "s#__version__ = \"$oldVersion\"#__version__ = \"$newVersion\"#g" python-shell/src/*/__init__.py
+
+    - name: Update gafferpy
+      run: |
+        newVersion=${{ github.event.inputs.version }}
+        curl -o spring-rest.jar https://repo.maven.apache.org/maven2/uk/gov/gchq/gaffer/spring-rest/$newVersion/spring-rest-$newVersion-exec.jar
+        java -Dgaffer.schemas=ui/example/road-traffic/schema/ -Dgaffer.storeProperties=ui/example/federated/federatedStore.properties -Dgaffer.graph.config=ui/example/federated/graphConfig.json -jar spring-rest.jar &
+        sleep 1m
+        python python-shell/src/generate.py
+        rm spring-rest.jar
```
Comment (GCHQDeveloper314) [inline .github/workflows/update-gaffer-version.yaml:46]: I wouldn't bother removing it.

File: .github/workflows/update-gaffer-version.yaml (hunk 1 above)
Comment (t92549) [reply .github/workflows/update-gaffer-version.yaml:49]: Yes, but I think the commit is associated with the account that activated the action

File: .github/workflows/update-gaffer-version.yaml (hunk 2 above)
Comment (t92549) [reply .github/workflows/update-gaffer-version.yaml:46]: It's either that or it gets put into the PR by the action

**Review by GCHQDeveloper314 — APPROVED** [review]

**Pull request merged**